sktime/forecasting/compose/_pipeline.py

```diff
@@ -202,10 +202,10 @@ def _update(self, y, X=None, update_params=True):
         self : an instance of self
         """
 
-        for step_idx, name, transformer in self._iter_transformers():
```

Comment:
@mloning I commented this block of code out and the stlforecaster passes the tests. As well as the transformedTargetForecaster. 
My understanding of this block of code is : 
we want to update the transformers if they have update methods. I want to know why this is necessary. Please, let me know if I got this wrong. thank you

Reply:
this might be important to keep - logically, the transformers should update.

What this indicates though is that we may want to look closely at the transformers and do a similar refactor.